tests/Doctrine/Tests/DBAL/Driver/Mysqli/MysqliConnectionTest.php

```diff
@@ -50,34 +50,5 @@ public function testRestoresErrorHandlerOnException()
         restore_error_handler();
     }
 
-    /**
-     * @dataProvider secureMissingParamsProvider
-     */
-    public function testThrowsExceptionWhenMissingMandatorySecureParams(array $secureParams)
```

Comment:
Could you please modify this test to make it work in the new behaviour (with the proper assertions), instead of simply removing it? Without that we don't have anything testing that `setSecureConnection()` works correctly.

Reply:
@lcobucci 

There are two path to this block of code. The first is the parameter validation

```
        if (! isset($params['ssl_key']) &&
            ! isset($params['ssl_cert']) &&
            ! isset($params['ssl_ca']) &&
            ! isset($params['ssl_capath']) &&
            ! isset($params['ssl_cipher'])
        ) {
            return;
        }
```
Which translate to "don't do anything to the connection if there are missing parameters". Because this modify the connection, I am not sure how this can be tested with the actual state of the code. We would need to test that `ssl_set` function is not called on `_conn` variable .

The other block has the same problem (make sure that `ssl_set` is called). To properly test the code, we either need a mysql server or rewrite this code so that the _conn is somehow mockable.

```
$this->_conn->ssl_set(
            $params['ssl_key']    ?? null,
            $params['ssl_cert']   ?? null,
            $params['ssl_ca']     ?? null,
            $params['ssl_capath'] ?? null,
            $params['ssl_cipher'] ?? null
        );
```

While I highly encourage unit-testing, there are some problem that are not easily testable like this one. It highly depends on a third party program (mysql) and the code is not written for easy unit-testing. Also, the setSecureConnection is a private function which renders testing even more harder. We might consider mocking `mysqli_init` but this seems hackish to me, I don't know.

The actual state of the code makes unit-testing very hard. A rewrite would be required to test everything this class does. Integrated test would be more suitable (test against a real mysql server).

Reply:
@nvanheuverzwijn you're correct, thanks! Let's ship it, then 😄

Reply:
Thank you for your time ! Cheers !